ubcpi/ubcpi.py

```diff
@@ -0,0 +1,573 @@
+    changes to the question options and may not change the order of the options,
+    add or delete options
+    """
+
+    # the display name that used on the interface
+    display_name = String(default="Peer Instruction")
+
+    question_text = Dict(
+        default={'text': 'What is the answer to life, the universe and everything?',
+                 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''},
+        scope=Scope.content,
+        help="The question the students see. This question appears above the possible answers which you set below. "
+             "You can use text, an image or a combination of both. If you wish to add an image to your question, press "
+             "the 'Add Image' button."
+    )
+
+    options = List(
+        # default=['Default Option 1', 'Default Option 2'], scope=Scope.content,
+        default=[
+            {'text': '21', 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''},
+            {'text': '42', 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''},
+            {'text': '63', 'image_url': '', 'image_position': 'below', 'show_image_fields': 0, 'image_alt': ''}
+        ],
+        scope=Scope.content,
+        help="The possible options from which the student may select",
+    )
+
+    rationale_size = Dict(
+        default={'min': 1, 'max': MAX_RATIONALE_SIZE}, scope=Scope.content,
+        help="The minimum and maximum number of characters a student is allowed for their rationale.",
+    )
+
+    correct_answer = Integer(
+        default=1, scope=Scope.content,
+        help="The correct option for the question",
+    )
+
+    correct_rationale = Dict(
+        default={'text': "Specify the rationale to explain the correct answer"}, scope=Scope.content,
+        help="The feedback for student for the correct answer",
+    )
+
+    stats = Dict(
+        default={'original': {}, 'revised': {}}, scope=Scope.user_state_summary,
+        help="Overall stats for the instructor",
+    )
+    seeds = List(
+        default=[], scope=Scope.content,
+        help="Instructor configured examples to give to students during the revise stage.",
+    )
+
+    # sys_selected_answers dict format:
+    # {
+    #     option1_index: {
+    #         'student_id1': { can store algorithm specific info here },
+    #         'student_id2': { can store algorithm specific info here },
+    #         ...
+    #     }
+    #     option2_index: ...
+    # }
+    sys_selected_answers = Dict(
+        default={}, scope=Scope.user_state_summary,
+        help="System selected answers to give to students during the revise stage.",
+    )
+
+    algo = Dict(
+        default={'name': 'simple', 'num_responses': '#'}, scope=Scope.content,
+        help="The algorithm for selecting which answers to be presented to students",
+    )
+
+    # Declare that we are part of the grading System
+    has_score = True
```

Comment:
Why do you want to be part of the grading system?

You don't give instructors the ability to set a different level of score.

Further, every time you do grading runs (generating certificates or pulling user data), or users visit the progress page, every single PI XBlock will be instantiated _per student_ to get their state and score.

If you set this to False, our most critical performance concerns evaporate --- the progress page is the single longest load page in the system, so we're very conscious of adding more time to this load.